### PR TITLE
INTC-97 Create a Jenkins Plugin Trend Graph UI (frontend)

### PR DESCRIPTION
**Pull request by Mykyta**
#### Description
Add the Nexus IQ chart under the existing **Latest Nexus IQ Policy Evaluation** plugin for Jenkins

#### Links
JIRA: [INTC-97](https://issues.sonatype.org/browse/INTC-97)
Build: 

#### Screencast
https://youtu.be/3zoXZDNJJNY

### CODE REVIEW

File: src/main/resources/org/sonatype/nexus/ci/iq/Messages.properties [hunk 1]
```diff
@@ -36,6 +36,7 @@ IqPolicyEvaluation.EvaluationFailed=IQ Server evaluation of application {0} fail
 IqPolicyEvaluation.EvaluationWarning=IQ Server evaluation of application {0} detected warnings
 IqPolicyEvaluation.NodeContextRequired=nexusPolicyEvaluation step requires a node context. Please specify an agent or a node block
 
+IqPolicyEvaluation.ChartName=Trends for Nexus IQ Policy Evaluation
```
Comment (rjackson64840) [inline src/main/resources/org/sonatype/nexus/ci/iq/Messages.properties]: 0 : have Justin sign off on verbiage

File: src/main/resources/org/sonatype/nexus/ci/config/Messages.properties [hunk 2]
```diff
@@ -24,3 +24,4 @@ Configuration.TestConnection=Test Connection
 NxiqConfiguration.DisplayName=Nexus IQ Server
 NxiqConfiguration.ConnectionSucceeded=Nexus IQ Server connection succeeded ({0} applications)
 NxiqConfiguration.ConnectionFailed=Nexus IQ Server connection failed
+NxiqConfiguration.PluginId=nexus-jenkins-plugin
```
Comment (rjackson64840) [inline src/main/resources/org/sonatype/nexus/ci/config/Messages.properties]: -1 : this is not a message constant

File: src/main/resources/org/sonatype/nexus/ci/iq/PolicyEvaluationProjectAction/jobMain.groovy [hunk 3]
```diff
@@ -15,8 +15,14 @@ package org.sonatype.nexus.ci.iq.PolicyEvaluationProjectAction
...
 import org.sonatype.nexus.ci.iq.PolicyEvaluationProjectAction
+import lib.JenkinsTagLib
+import lib.LayoutTagLib
 
-def t = namespace(lib.JenkinsTagLib)
+def t = namespace(JenkinsTagLib.class)
+def l = namespace(LayoutTagLib.class)
+
+def pluginId = org.sonatype.nexus.ci.config.Messages.NxiqConfiguration_PluginId()
+def pluginResourcePath = "${rootURL}/plugin/${pluginId}"
```
Comment (rjackson64840) [inline src/main/resources/org/sonatype/nexus/ci/iq/PolicyEvaluationProjectAction/jobMain.groovy]: -1 : no need to obfuscate the usage of the resource path - just put it inline like the others - that way folks can see it and visually confirm it is correct

File: src/main/resources/org/sonatype/nexus/ci/config/Messages.properties (hunk 2 above)
Comment (Mykyta) [reply src/main/resources/org/sonatype/nexus/ci/config/Messages.properties]: I'll move to const var to file where I use it.

File: src/main/resources/org/sonatype/nexus/ci/iq/PolicyEvaluationProjectAction/jobMain.groovy (hunk 3 above)
Comment (Mykyta) [reply src/main/resources/org/sonatype/nexus/ci/iq/PolicyEvaluationProjectAction/jobMain.groovy]: Ok, then I'll get rid of these constants.

File: src/main/webapp/src/chart/iqChart.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    height: CHART_HEIGHT,
+    width: CHART_WIDTH,
+    type: 'line',
+    zoom: {
+      enabled: false
+    },
+    events: {
+      markerClick: function(event, series, dataPoint) {
+        const buildNumber = getXAxisLabels()[dataPoint.dataPointIndex];
+        window.location.href = buildNumber.toString();
```
Comment (rjackson64840) [inline src/main/webapp/src/chart/iqChart.js]: 0 : does this even work?

Comment (Mykyta) [reply src/main/webapp/src/chart/iqChart.js]: Yes, you can see on https://youtu.be/3zoXZDNJJNY?t=14

File: src/main/webapp/src/chart/iqChart.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
+/*
```
Comment (rjackson64840) [inline src/main/webapp/src/chart/iqChart.js]: -1 : I don't believe this is the package name we want.

For 3rd party libraries, like the apex charting, I would put them here:  `../webapp/lib`

For our custom javascript I would put them here:  `../webapp/features/iq/charting`

Comment (Mykyta) [reply src/main/webapp/src/chart/iqChart.js]: Ok, I'll move it to `lib`

File: src/main/resources/org/sonatype/nexus/ci/iq/PolicyEvaluationProjectAction/jobMain.groovy [hunk 6]
```diff
@@ -16,7 +16,14 @@ import org.sonatype.nexus.ci.iq.Messages
...
 
-def t = namespace(lib.JenkinsTagLib)
+import jenkins.model.Jenkins
+import lib.JenkinsTagLib
+import lib.LayoutTagLib
+
+def t = namespace(JenkinsTagLib.class)
+def l = namespace(LayoutTagLib.class)
+
+def pluginResourcePath = "${Jenkins.instance.rootUrl}/plugin/nexus-jenkins-plugin"
```
Comment (rjackson64840) [inline src/main/resources/org/sonatype/nexus/ci/iq/PolicyEvaluationProjectAction/jobMain.groovy]: -1 : why do we need the root URL prefix?  aren't urls already relative to the root?

File: src/main/resources/org/sonatype/nexus/ci/iq/PolicyEvaluationProjectAction/jobMain.groovy [hunk 7]
```diff
@@ -113,4 +120,13 @@ if (action) {
   table(class: 'iq-job-main-table') {
     t.summary(icon: '/plugin/nexus-jenkins-plugin/images/48x48/nexus-iq.png', policyUI << policyCss)
   }
+
+  table(class: 'iq-job-main-table') {
+    l.main_panel() {
+      div(id: 'iqChart')
+      script(src: "${pluginResourcePath}/lib/apexcharts.js")
```
Comment (rjackson64840) [inline src/main/resources/org/sonatype/nexus/ci/iq/PolicyEvaluationProjectAction/jobMain.groovy]: 0 : I would just put the path inline

File: src/main/resources/org/sonatype/nexus/ci/iq/PolicyEvaluationProjectAction/jobMain.groovy (hunk 6 above)
Comment (Mykyta) [reply src/main/resources/org/sonatype/nexus/ci/iq/PolicyEvaluationProjectAction/jobMain.groovy]: `script(src: "/lib/apexcharts.js")` refer to http://localhost:8090/features/iq/charting/iqChart.js
so we get 404.
`Jenkins.instance.rootUrl` refer to http://localhost:8090/jenkins
You can see, the same across the whole project https://github.com/jenkinsci/nexus-platform-plugin/blob/5b0aa4eb90886977d1859fa025dc34e44b3e557a/src/main/resources/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorBuildStep/config.groovy#L42

File: src/main/resources/org/sonatype/nexus/ci/iq/PolicyEvaluationProjectAction/jobMain.groovy (hunk 7 above)
Comment (Mykyta) [reply src/main/resources/org/sonatype/nexus/ci/iq/PolicyEvaluationProjectAction/jobMain.groovy]: Ok

**Comment by Mykyta** [comment]
> +1 : how does the data actually flow into the chart?

I pass them via the Groovy in script section and get in JS by 
`document.currentScript.getAttribute` as JSON object
You can see it in (INTC-96)[https://issues.sonatype.org/browse/INTC-96]

**Comment by Mykyta** [comment]
no valid any mode

**Pull request closed without merging**